cloud-server/src/main/java/MessagesHandler.java

```diff
@@ -4,60 +4,81 @@
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.*;
-import java.util.concurrent.ConcurrentLinkedDeque;
 import java.util.stream.Collectors;
 
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.channel.SimpleChannelInboundHandler;
+import messages.*;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import tools.ObjectWriter;
 
-public class MessagesHandler extends SimpleChannelInboundHandler<ExchangeMessage> {
+public class MessagesHandler extends SimpleChannelInboundHandler<ExchangeMessage> implements ObjectWriter {
 
-    private static final ConcurrentLinkedDeque<ChannelHandlerContext> clients = new ConcurrentLinkedDeque<>();
     private static final Logger LOG = LoggerFactory.getLogger(MessagesHandler.class);
+    private static final int SEND_BUFFER_LENGTH = 65536;
 
     private Path serverPath = Paths.get(System.getProperty("user.home") + "/tmp").toAbsolutePath().normalize();
     private Path newPath = serverPath;
-    private static HashSet<String> processing = new HashSet<>();
+    private AuthMessage user = null;
 
     private static int cnt = 0;
 
     @Override
     public void channelActive(ChannelHandlerContext ctx) {
-        clients.add(ctx);
-        cnt++;
         LOG.info("current path is " + serverPath);
     }
 
+    public void writeObject(Object stream, Object obj) {
+        ((ChannelHandlerContext)stream).writeAndFlush(obj);
+    }
+
     @Override
     protected void channelRead0(ChannelHandlerContext ctx, ExchangeMessage msg) throws Exception {
         LOG.info("Received " + msg.getClass().getName());
-        switch (msg.getClass().getName()) {
-            case "FileMessage" : processFileMessage((FileMessage) msg);
-                                 break;
-            case "CommandMessage": processCommandMessage((CommandMessage) msg);
+        if (msg instanceof AuthMessage) {
+            if (processAuthMessage((AuthMessage) msg)) {
+                user = (AuthMessage)msg;
+                ctx.writeAndFlush(new TextMessage("Authorized"));
+            }
+        }
 
+        if (user == null) {
+            LOG.warn("Non Authorized user");
+            return;
+        }
 
+        if (msg instanceof FileMessage) {
+            processFileMessage((FileMessage) msg);
+            ctx.writeAndFlush(new TextMessage(String.format("Received %s part",((FileMessage)msg).getName())));
+        } else if (msg instanceof ListCommandMessage || msg instanceof BoolCommandMessage) {
+            processCommandMessage((CommandMessage) msg);
+            ctx.writeAndFlush(msg);
+        } else if (msg instanceof RequestFileMessage) {
+            FileMessage.sendByStream(newPath.resolve(((RequestFileMessage)msg).getParam()), SEND_BUFFER_LENGTH, this, ctx);
         }
     }
 
+    private boolean processAuthMessage(AuthMessage msg) {
+        //TODO: Check Authorization
+        return (msg.getLogin() == "cloud_user" && msg.getPasswordHash() == "Abc12#".hashCode());
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.